pkg/rbd/controllerserver.go

```diff
@@ -156,7 +186,13 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 	}
 	defer cs.VolumeLocks.Release(req.GetName())
 
-	found, err := checkVolExists(ctx, rbdVol, cr)
+	err = rbdVol.Connect(cr)
+	if err != nil {
+		klog.Errorf(util.Log(ctx, "failed to connect to volume %v: %v"), rbdVol.RbdImageName, err)
+		return nil, err
```

Comment:
Connects just return Errorf. need to return grpc error code and message

Reply:
done with the next update